cmd/admin-config-setkeys.go

```diff
@@ -0,0 +1,157 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/minio/pkg/madmin"
+)
+
+var adminConfigSetKeysCmd = cli.Command{
+	Name:   "setkeys",
+	Usage:  "Set partial Minio server/cluster configuration parameters using key/value pairs.",
+	Before: setGlobalsFromContext,
+	Action: mainAdminConfigSetKeys,
+	Flags:  globalFlags,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Set a single Minio server/cluster configuration entry.
+     $ {{.HelpName}} myminio 
+
+  2. Set two Minio server/cluster configuration entries.
+     $ {{.HelpName}} myminio logger.console.enabled=true logger.http.minio_one.enabled=true
+
+  3. Set multiple Minio server/cluster configuration entries.
+     $ {{.HelpName}} myminio logger.http.minio_one='{
+                "enabled": true,
+                "endpoint": "https://one.minio.io/logger/<some-jwt>"
+            }'
+`,
+}
+
+// configSetKeysMessage container to hold locks information.
+type configSetKeysMessage struct {
+	Status                   string `json:"status"`
+	setKeysConfigStatus      bool
+	SetKeysConfigNodeSummary []madmin.NodeSummary `json:"nodesSummary"`
+}
+
+// String colorized service status message.
+func (u configSetKeysMessage) String() (msg string) {
+	// For each node, print if set config was successful or not
+	for _, s := range u.SetKeysConfigNodeSummary {
+		if s.ErrSet {
+			// An error occurred when setting the new config
+			msg += console.Colorize("SetKeysConfigFailure",
+				fmt.Sprintf("Failed to apply the new configuration to `%s` (%s).", s.Name, s.ErrMsg))
+		} else {
+			msg += console.Colorize("SetKeysConfigSuccess",
+				fmt.Sprintf("New configuration applied to `%s` successfully.", s.Name))
+		}
+		msg += "\n"
+	}
+	// Print the general set config status
+	if u.setKeysConfigStatus {
+		msg += console.Colorize("SetKeysConfigSuccess",
+			"Setting new Minio configuration file has been successful.\n")
+	} else {
+		msg += console.Colorize("SetKeysConfigFailure",
+			"Setting new Minio configuration file has failed.\n")
+	}
+	return
+}
+
+// JSON jsonified service status Message message.
+func (u configSetKeysMessage) JSON() string {
+	if u.setKeysConfigStatus {
+		u.Status = "success"
+	} else {
+		u.Status = "error"
+	}
+	statusJSONBytes, e := json.Marshal(u)
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(statusJSONBytes)
+}
+
+// checkAdminConfigSetKeysSyntax - validate all the passed arguments
+func checkAdminConfigSetKeysSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 {
+		cli.ShowCommandHelpAndExit(ctx, "set", 1) // last argument is exit code
```

Comment:
This should be `cli.ShowCommandHelpAndExit(ctx, "setkeys", 1) // last argument is exit code`